### PR TITLE
port of Java #254

### PR DESCRIPTION
**Pull request by kwhopper**
Port of drewnoakes/metadata-extractor#254

### CODE REVIEW

File: MetadataExtractor/Formats/Gif/GifReader.cs [hunk 1]
```diff
@@ -129,6 +134,7 @@ private static IEnumerable<Directory> ReadGifDirectoriesInternal(SequentialReade
                         // Anything other than these types is unexpected.
                         // GIF87a spec says to keep reading until a separator is found.
                         // GIF89a spec says file is corrupt.
+                        yield return new ErrorDirectory("Unknown gif block marker found.");
```
Comment (drewnoakes) [inline MetadataExtractor/Formats/Gif/GifReader.cs]:
```suggestion
                        yield return new ErrorDirectory("Unknown GIF block marker found.");
```

File: MetadataExtractor/Formats/Gif/GifReader.cs [hunk 2]
```diff
@@ -80,13 +80,18 @@ private static IEnumerable<Directory> ReadGifDirectoriesInternal(SequentialReade
 
             if (header.HasError)
                 yield break;
-
-            // Skip over any global colour table
-            if (header.TryGetInt32(GifHeaderDirectory.TagColorTableSize, out int globalColorTableSize))
+            
+            if(header.TryGetBoolean(GifHeaderDirectory.TagHasGlobalColorTable, out bool hasGlobalColorTable))
```
Comment (drewnoakes) [inline MetadataExtractor/Formats/Gif/GifReader.cs]:
```suggestion
            if (header.TryGetBoolean(GifHeaderDirectory.TagHasGlobalColorTable, out bool hasGlobalColorTable))
```

File: MetadataExtractor/Formats/Gif/GifReader.cs [hunk 3]
```diff
@@ -80,13 +80,18 @@ private static IEnumerable<Directory> ReadGifDirectoriesInternal(SequentialReade
...
-                reader.Skip(3 * globalColorTableSize);
+                // Skip over any global colour table
+                if (hasGlobalColorTable && header.TryGetInt32(GifHeaderDirectory.TagColorTableSize, out int globalColorTableSize))
+                {
+                    // Colour table has R/G/B byte triplets
+                    reader.Skip(3 * globalColorTableSize);
+                }
             }
+            else
+                yield return new ErrorDirectory("GIF did not have hasGlobalColorTable bit.");
```
Comment (drewnoakes) [inline MetadataExtractor/Formats/Gif/GifReader.cs]:
```suggestion
            {
                yield return new ErrorDirectory("GIF did not have hasGlobalColorTable bit.");
            }
```

File: MetadataExtractor/Formats/Gif/GifReader.cs (hunk 1 above)
Comment (kwhopper) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: gtg

File: MetadataExtractor/Formats/Gif/GifReader.cs (hunk 2 above)
Comment (kwhopper) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: gtg

File: MetadataExtractor/Formats/Gif/GifReader.cs (hunk 3 above)
Comment (kwhopper) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: ... and gtg

**Pull request merged**